src/rpp/rpp/operators/combine_latest.hpp

```diff
@@ -0,0 +1,173 @@
+    struct combine_latest_coordinator_t
+    {
+        explicit combine_latest_coordinator_t(TCombiner combiner) : m_combiner(combiner) {}
+
+        // Not copyable nor movable.
+        combine_latest_coordinator_t(const combine_latest_coordinator_t&) = delete;
+        combine_latest_coordinator_t(combine_latest_coordinator_t&&) = delete;
+
+        template<size_t I>
+        void offer_next_value(auto&& value, const auto& subscriber)
+        {
+            if (!subscriber.is_subscribed()) return;
+
+            std::scoped_lock<std::mutex> lock{m_mutex};
+            std::get<I>(m_values) = std::forward<decltype(value)>(value);
+
+            // If all the optional(s) have value, dispatch to downstream
+            if (has_values_for_observer())
+            {
+                subscriber.on_next(combine_cached_values());
+            }
+        }
+
+        void offer_completed(const auto& subscriber)
+        {
+            if (!subscriber.is_subscribed()) return;
+
+            std::scoped_lock<std::mutex> lock{m_mutex};
+
+            if (++m_completed_count == m_total_completed_count)
+            {
+                subscriber.on_completed();
+            }
+        }
+
+    private:
+        TCombiner m_combiner;
+
+        std::mutex m_mutex;
+        std::tuple<std::optional<SourceValueType>, std::optional<OtherValueTypes>...> m_values{};
+        size_t m_completed_count{0};
+
+        const std::size_t m_total_completed_count = 1 + sizeof...(OtherValueTypes);
+
+        bool has_values_for_observer()
+        {
+            return std::apply([](const auto&... cached_values)
+            {
+                // Variadic expansion used
+                return (cached_values.has_value() && ...);
+            }, m_values);
+        }
+
+        auto combine_cached_values()
+        {
+            return std::apply([&](const auto&... cached_values)
+            {
+                // Variadic expansion used
+                return m_combiner(cached_values.value()...);
+            }, m_values);
+        }
+    };
+
+    /**
+     * \brief Templated helper function for subscribing to variadic 'other' observables.
+     *
+     * \param observables is a variadic tuple. The first observable is the current observable and rest are the other observables.
+     * \param subscriber is the downstream subscriber.
+     * \param coordinator manages the cache of emission from the observables and coordinate dispatching.
+     */
+    template<size_t...I>
+    static void subscribe_other_observables(std::index_sequence<I...>, // Used in compile time for variadic expansion
+                                            const auto& observables,
+                                            const auto& subscriber,
+                                            const auto& coordinator)
+    {
+        // +1 because the first element in tuple is the current observable, and you want to subscribe to the 'other' observables.
```

Comment:
It is not true, because tuple contains only "other" observables, while subscribe on "current" happens inside `operator()` via returning subscriber for "current"

Reply:
I don't quite understand. I design the tuple to include this observable and other observable.

Reply:
Oh, I see, my miss-understanding... But it looks really hard to understand =)

Reply:
Haha, my bad. I know it's hard especially the behavior of this is quite similar to `with_latest_from()`. But it does make sense to have a tuple for this observable + other observables.